tests/file_writer/test_file_writer.py

```diff
@@ -117,11 +112,289 @@ def test_FileWriterProcess_soft_stop_not_allowed_if_incoming_data_still_in_queue
+    GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]["auto_delete_local_files"] = auto_delete
+    GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]["auto_upload_on_completion"] = auto_upload
+    this_command = copy.deepcopy(GENERIC_UPDATE_CUSTOMER_SETTINGS)
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(this_command, from_main_queue)
+    invoke_process_run_and_check_errors(file_writer_process)
+
+    assert (len(mocked_delete_files.call_args_list) > 0) is auto_delete
+    assert (len(spied_thread_start.call_args_list) > 0) is auto_upload
+
+
+@pytest.mark.parametrize(
+    "move_called, thread_error",
+    [(True, False), (False, True)],
+)
+def test_FileWriterProcess__exits_status_function_when_newly_failed_files_errors_or_passes(
+    four_board_file_writer_process, move_called, thread_error, mocker
+):
+    file_writer_process = four_board_file_writer_process["fw_process"]
+
+    mocked_shutil = mocker.patch.object(shutil, "move", autospec=True)
+    mocked_thread = mocker.patch.object(file_uploader, "ErrorCatchingThread", autospec=True)
+    mocker.patch.object(mocked_thread.name, "is_alive", autospec=True, return_value=False)
+    mocker.patch.object(mocked_thread.name, "errors", autospec=True, return_value=thread_error)
+    mocker.patch.object(mocked_thread.name, "get_upload_status", autospec=True)
+    mocker.patch.object(mocked_thread.name, "join", autospec=True)
+
+    thread_dict = {
+        "failed_upload": True,
+        "customer_account_id": "test_customer_id",
+        "thread": mocked_thread.name,
+        "auto_delete": False,
+        "file_name": "test_filename",
+    }
+
+    file_writer_process._upload_threads_container.append(thread_dict)  # pylint: disable=protected-access
+    file_writer_process._check_upload_statuses()  # pylint: disable=protected-access
+
+    assert (len(mocked_shutil.call_args_list) > 0) is move_called
+    assert len(file_writer_process._upload_threads_container) == 0  # pylint: disable=protected-access
+
+
+def test_FileWriterProcess__kicks_off_upload_thread_and_appends_to_container_with_specified_dict_values(
+    four_board_file_writer_process, mocker
+):
+
+    file_writer_process = four_board_file_writer_process["fw_process"]
+    from_main_queue = four_board_file_writer_process["from_main_queue"]
+
+    GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]["auto_upload_on_completion"] = True
+    GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]["auto_delete_local_files"] = False
+    this_command = copy.deepcopy(GENERIC_UPDATE_CUSTOMER_SETTINGS)
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(this_command, from_main_queue)
+
+    mocked_check_status = mocker.patch.object(file_writer_process, "_check_upload_statuses", auto_spec=True)
+    mocker.patch.object(file_uploader, "ErrorCatchingThread", autospec=True)
+
+    invoke_process_run_and_check_errors(file_writer_process)
+    # pylint: disable=protected-access
+    assert (
```

Comment:
If you wanted to, you could make a getter method for `_upload_threads_container` so you don't have to disable pylint warnings. When I do that, I typically say in the docstring that the method is specifically for use in unit testing. Not sure which way is better TBH